privacyidea/lib/token.py

```diff
@@ -1908,14 +1912,27 @@ def check_realm_pass(realm, passw, options=None):
     #  of an inactive token.
     tokenobject_list = get_tokens(realm=realm, assigned=True, active=True)
     if not tokenobject_list:
-        res = False
-        reply_dict["message"] = "There is no active and assigned token in " \
-                                "this realm"
+        reply_dict["message"] = "There is no active and assigned token in this realm"
+        return False, reply_dict
     else:
-        res, reply_dict = check_token_list(tokenobject_list, passw,
-                                           options=options,
-                                           allow_reset_all_tokens=False)
-    return res, reply_dict
+        # reduce tokens by type
+        if options:
+            if options.get('include_types'):
```

Comment:
*check_realm_pass* is very 4eyes token specific. It is not used anywhere else.
*exclude_types* and *include_types* are also very 4eyes token specific. *include_types* is actually not used anywhere - but might be in the future - nice idea.
The paramter *options* is usually data that is passed from the upper REST API level, so this might lead to confusion, that *exclude_types* could be something that is passed via the REST API (At least it had cofused me).

I would suggest to add dedicated keyword parameters for *exclude_types* and *include_types*, so that it is clear, that this data does not origin from the REST API level.

~~~~python
def check_realm_pass(realm, passw, options=None, exclude_types=None, include_types=None)
~~~~

Reply:
Will do.

Reply:
Done in 97217faa